### PR TITLE
Feat/routinely updated field widget

### PR DESCRIPTION
**Pull request by Sepehr-Sobhani**
[ZH Card](https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1382)

### CODE REVIEW

File: app/components/ProjectRevision/routinelyUpdatedFieldFactory.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  // mutationFnHook: () => [
+  //   (config: UseDebouncedMutationConfig<MutationParameters>) => Disposable,
+  //   boolean
+  // ] //TODO: bring this back when we have the mutation
+  // fieldNameToUpdate: string, //TODO: bring this back when we have the mutation
+  // formDataTableName: string //TODO: bring this back when we have the mutation
+): React.FC<WidgetProps> => {
+  const WrappedWidget: React.FC<WidgetProps> = (props) => {
+    // helper function to compare the initial value to the current value
+    const valueUnchanged = useMemo(
```
Comment (BCerki) [inline app/components/ProjectRevision/routinelyUpdatedFieldFactory.tsx:36]: I'm not sure about the name `valueUnchanged` This is the value that shows up before the user touches anything in the widget? I think `initalValue` like you've got in the comment above might be clearer

**Review by BCerki — COMMENTED** [review]
I see what you mean about wanting to have the mutation in place before merging, but there's [another card](https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1392) for linking the widget and the mutation, and that will include tests. So I think you've done everything your card requires.

File: app/components/ProjectRevision/routinelyUpdatedFieldFactory.tsx (hunk 1 above)
Comment (Sepehr-Sobhani) [reply app/components/ProjectRevision/routinelyUpdatedFieldFactory.tsx:36]: This is a mini helper function to compare the untouched value with the current value and does some job based on that. I think `initialValue` would conflict with the existing `initialValue` state but any other name would be acceptable :) I'm not good at naming vars :D

**Comment by Sepehr-Sobhani** [comment]
> I see what you mean about wanting to have the mutation in place before merging, but there's [another card](https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1392) for linking the widget and the mutation, and that will include tests. So I think you've done everything your card requires.

Oh true, I didn't consider that card. so I think this card can go forward.

File: app/components/ProjectRevision/routinelyUpdatedFieldFactory.tsx (hunk 1 above)
Comment (BCerki) [reply app/components/ProjectRevision/routinelyUpdatedFieldFactory.tsx:36]: Oh right, `initialValue` is already taken. Maybe something like... `committedValue`? `previouslyCommittedValue`? But I don't think it's a big deal or a necessary change

**Review by BCerki — DISMISSED** [review]

**Review by BCerki — APPROVED** [review]

**Pull request merged**